tests/testthat/test-app_pws.R

```diff
@@ -0,0 +1,300 @@
+  368.846473150198, 5.47945205479452, character(0), character(0)
+)
+
+dbl_click_vals <- c(
+  2.95684692272772, 2.34130469829519,
+  789, 247, 789, 247, 1, 1, "1:5", "1:5", 1, 5, 1, 5, 28.0413099315069, 1583.52054794521,
+  368.846473150198, 5.47945205479452, character(0), character(0)
+)
+
+click_vals <- c(
+  2.95941847746291, 1.9450111668517,
+  790, 283, 790, 283, 1, 1, "1:5", "1:5", 1, 5, 1, 5, 28.0413099315069, 1583.52054794521,
+  368.846473150198, 5.47945205479452, character(0), character(0)
+)
+
+# one app opens for all tests
+pws <- app_pws()
+app <- AppDriver$new(pws, name = pws, height = 937, width = 1619)
+
+# tests plot click functionalities
+testthat::test_that("plot_with_settings: click functionalities ggplot2", {
+  skip_on_cran()
+  skip_on_ci()
+
+  # hovering
+  app$set_inputs(
+    `plot_with_settings-plot_hover` = hover_vals,
+    allow_no_input_binding_ = TRUE, priority_ = "event"
+  )
+
+  # double click
+  app$set_inputs(
+    `plot_with_settings-plot_dblclick` = dbl_click_vals,
+    allow_no_input_binding_ = TRUE, priority_ = "event"
+  )
+
+  # click
+  app$set_inputs(
+    `plot_with_settings-plot_click` = click_vals,
+    allow_no_input_binding_ = TRUE, priority_ = "event"
+  )
+
+  # brushing
+  app$set_inputs(
+    `plot_with_settings-plot_brush` = brush_vals, allow_no_input_binding_ = TRUE
+  )
+
+  vals <- app$get_values()
+
+  # testing clicking / hovering / brushing
+  test_hover <- shiny::isolate(vals$export$plot_data$hover())
+  testthat::expect_equal(
+    test_hover,
+    hover_vals
+  )
+
+  test_dblclick <- shiny::isolate(vals$export$plot_data$dblclick())
+  testthat::expect_equal(
+    test_dblclick,
+    dbl_click_vals
+  )
+
+  test_click <- shiny::isolate(vals$export$plot_data$click())
+  testthat::expect_equal(
+    test_click,
+    click_vals
+  )
+
+  test_brush <- shiny::isolate(vals$export$plot_data$brush())
+  testthat::expect_equal(
+    test_brush,
+    brush_vals
+  )
+
+  # reset brush to character(0) for next tests
+  app$set_inputs(
+    `plot_with_settings-plot_brush` = character(0),
+    allow_no_input_binding_ = TRUE
+  )
+})
+
+# test output that is returned (reactives and graphic encoded in base64)
+testthat::test_that("plot_with_settings: output is returned", {
+  skip_on_cran()
+  skip_on_ci()
+
+  vals <- app$get_values()
+
+  # check if outputs are reactive
+  for (react_i in vals$export$"plot_data") {
+    testthat::expect_true(is(react_i, "reactive"))
+  }
+
+  # check if output is desired plot
+  testthat::expect_type(
+    vals$output$`plot_with_settings-plot_main`$src, "character"
+  )
+  # nolint start
+  testthat::expect_true(
+    grepl("data:image/png;base64,",
+      vals$output$`plot_with_settings-plot_main`$src,
+      fixed = TRUE
+    )
+  )
+  # nolint end
+})
+
+# download plots. expect_download() might not be stable, hence we test
+# setting inputs and plot name changes
+testthat::test_that("plot_with_settings: download functionality ggplot2", {
```

Comment:
This test doesn't test anything, it just changes the values of the inputs and checks if the inputs are set - they are set, but by the test directly. We should rather test:
- if clicks in this module returns proper and sufficient information to download an image
- and if the image is downloaded with expected name, dimensions, file type. 

Name is also imprecise as download module has nothing to do with the plot type - but `print_plot` has - which can be tested in the separete test_that